crates/shared/src/ethrpc/multicall.rs

```diff
@@ -0,0 +1,426 @@
+//! Multicall encoding and decoding.
+
+use crate::ethrpc::dummy;
+use contracts::support::Multicall;
+use ethcontract::{
+    errors::ExecutionError,
+    tokens::{self, Tokenize as _},
+};
+use hex_literal::hex;
+use lazy_static::lazy_static;
+use std::iter;
+use web3::{
+    self,
+    api::Eth,
+    ethabi::{self, ParamType, Token},
+    types::{AccessList, BlockId, Bytes, CallRequest, H160, U256, U64},
+    Transport,
+};
+
+/// A single call in a multicall batch.
+#[derive(Clone, Debug, Default, Eq, PartialEq)]
+pub struct Call {
+    /// The address to call.
+    pub to: H160,
+
+    /// Optional gas limit to use for the call.
+    pub gas: Option<U256>,
+
+    /// Optional value to use for the call.
+    pub value: Option<U256>,
+
+    /// Data to use for the call.
+    pub data: Bytes,
```

Comment:
Should be plain `Vec<u8>`. `ethabi::Bytes` type is only used for serialization.

Reply:
Good point. Changed to use `Vec<u8>` at the interface boundries (both the `Call` type and the `multicall` return).